### PR TITLE
Add GetState for VirtualMachineSpec

### PR DESCRIPTION
**Pull request by johnsonshih**
get vm properties using fresh computeSystem

### CODE REVIEW

File: vm.go [hunk 1]
```diff
@@ -743,27 +702,48 @@ func (vm *VirtualMachineSpec) AssignDevice(ctx context.Context, deviceID string)
...
+		stopped = true
+		return
+	}
+	return properties.State, properties.Stopped, nil
 }
 
-func (vm *VirtualMachineSpec) OS() string {
+func (vm *VirtualMachineSpec) OS() (string, error) {
 	if vm.gc == nil {
-		return vm.system.OS()
```
Comment (jagadishmurugan) [inline vm.go:764]: what's the motivation for this change? why revert the changes?

Comment (johnsonshih) [reply vm.go:764]: I had put the reason in comment

 

        // The properties of hcs.System are set when hcs.System is created and never refresh
        // cannot use vm.system as vm.system was created at vm.Create() and doesn't have up-to-date properties
        // get the properties from a fresh hcs.System

**Review by jagadishmurugan — APPROVED** [review]
